### ISSUE TITLE
You cannot see un-pushed commits if you restart the application

### ISSUE THREAD

**Issue by trstndvll** [uvicgse/project-virtual-team-1#70]
If you edit a file, then commit those changes, and restart the app, there is no visual indication that there are commits waiting to be pushed.

**Steps to reproduce:**
1. Open a repo in VisualGit
2. Edit a file and commit the changes
3. Close and re-open the application

**For example, GitHub Desktop does show waiting commits:**
![image](https://user-images.githubusercontent.com/12837282/58586345-b5ef7400-820f-11e9-86c6-0bc571ac2fe9.png)

This issue would be solved by adding similar functionality to VisualGit.

**Comment by sterlinglaird** [comment]
I guess a large part of the problem is that you cannot tell the difference between a remote and a local branch so you cannot tell when your local branch is behind. Other GUI's do this by displaying the remote as well as the local branch in the graph.

**Comment by trstndvll** [comment]
@sterlinglaird I agree there needs to be some visual indication that there is a difference between local and remote. We discussed with James that we were going to try and load in commits on the graph once a repo is opened, and then if any commits are not on remote, we just give them a different colour than the rest of the nodes.

I think that would be cleaner than adding another GUI component similar to what GitHub Desktop does, but we're going to have to figure out how feasible that is.

**Comment by trstndvll** [comment]
We have working functions that calculate the number of local commits and the number of remote commits, and then takes the difference through a separate function. That gives us the number of un-pushed commits. The problem we're running into right now is we are using an asynchronous function using `async.whilst` to find the remote commits. Combining that with outputting to global variables to do the difference calculation, we run into an issue where our difference function needs to be called twice in order to output correctly. Still looking into why this is the case.

**Comment by aeogunmuyiwa** [comment]
We have created a button for this functionality, called "info" and its located on the header of the application. However users have  to click the button twice to get the latest unpushed commits . We also included a note on hover on the button explaining this. This is as a result of the limitation of nodegit API.

**Comment by aeogunmuyiwa** [comment]
-We now update the number of unpushed commits without a click to the function
-the value is displayed below the pull request tab.
- the value is updated on a set interval of 30000ms

**Comment by trstndvll** [comment]
![Screenshot 2017-01-28 11 54 58](https://user-images.githubusercontent.com/12837282/59141369-96b7bb80-8960-11e9-942a-a5c50ff05c91.png)
Preview of the new placement for the un-pushed commits count

**Comment by aeogunmuyiwa** [comment]
https://slack-files.com/TJBDCRM6G-FK4PT3Y4B-a3bdba6669
- because of lots of update to master, we had issues resolving the merge conflicts as the upgrade to angular and file changes had occurred. 
- I cloned master and copied the unpushed commits functions, the main merge conflicts issue was coming from getAllPushedCommits() function so I rewrote it to match the create changes in the git.ts file. 
- I created a new branch called issue-70-unpushedcommits-pr where the current working code can be found .

**Comment by aeogunmuyiwa** [comment]
https://slack-files.com/TJBDCRM6G-FK4PT3Y4B-a3bdba6669
The link about contains the functionality image. 
- because of lots of update to master, we had issues resolving the merge conflicts as the upgrade to angular and file changes had occurred.
-  I cloned master and copied the unpushed commits functions, the main merge conflicts issue was coming from getAllPushedCommits() function so I rewrote it to match the create changes in the git.ts file.
-I created a new branch called issue-70-unpushedcommits-pr where the current working code can be found.
